### PR TITLE
Add test that verifies that IOUtils.readAsset() closes the stream. (#898)

### PR DESCRIPTION
**Pull request by pocmo**
Don't have a fix for bug #898. But as I wrote this test to verify something I thought I could just land it.

### CODE REVIEW

**Comment by pocmo** [comment]
Now I felt like quickly converting IOUtils to Kotlin while I'm in that file already. Will update the PR. :)

File: app/src/test/java/org/mozilla/focus/ext/AssetManagerTest.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+        val stream = Mockito.spy(ByteArrayInputStream("Hello".toByteArray()))
+
+        val assetManager = Mockito.mock(AssetManager::class.java)
+        Mockito.`when`(assetManager.open(ArgumentMatchers.anyString())).thenReturn(stream)
+
+        val context = Mockito.mock(Context::class.java)
+        Mockito.`when`(context.assets).thenReturn(assetManager)
+
+        // Now use our mock classes to call IOUtils.close()
```
Comment (mcomella) [inline app/src/test/java/org/mozilla/focus/ext/AssetManagerTest.kt]: nit: IOUtils -> AssetManager (fwiw, I feel like most of the comments here are redundant with the code)

Comment (pocmo) [reply app/src/test/java/org/mozilla/focus/ext/AssetManagerTest.kt]: Ops, too fast..

**Comment by pocmo** [comment]
.. and I need to fix this test. Pretty sure this passed before. Weird. :)

**Comment by pocmo** [comment]
Fixed the test and added another one that verifies that the JSON is transformed into a JSONObject correctly.

**Comment by pocmo** [comment]
Now findbugs is complaining:
> Self assignment of it in org.mozilla.focus.ext.AssetManagerKt.readJSONObject(AssetManager, String)
https://public-artifacts.taskcluster.net/O5kTi4O4Spi_lOKUgEum1g/0/public/findbugs/findbugs.html

This is a bit unfortunate as the code findbugs is complaining about is the result of compiling the Kotlin code.

**Comment by pocmo** [comment]
Seems to be a known problem in Kotlin 1.1:
https://discuss.kotlinlang.org/t/should-we-be-reporting-findbugs-issues/2105

According to the thread above this is fixed in Kotlin 1.2. Kotlin 1.2 Beta2 was released recently - so 1.2 might not be that far away.

**Comment by pocmo** [comment]
For now I added the @SuppressFBWarnings annotation with an explanation of the problem.

**Pull request merged**